src/ts/components/core/chip/Chip.tsx

```diff
@@ -59,11 +60,15 @@ const Chip = (props: Props) => {
     const onChange = (checked: boolean) => {
         setProps({ checked });
     };
+
+    const { chipOnClick } = React.useContext(ChipGroupContext) || {};
```

Comment:
Very nice @RenaudLN!

I wonder if we can use this to get rid of the `controlled` prop entirely (cc @BSd3v)? something like:
```js
const eventProps = {};
if (chipOnClick) {  // we have a ChipGroup context
    eventProps.onClick = chipOnClick;
} else {
    eventProps.checked = checked;
    eventProps.onChange = onChange;
}
return <MantineChip
    data-dash-is-loading={
        (loading_state && loading_state.is_loading) || undefined
    }
    {...eventProps}
    {...others}
>
    {children}
</MantineChip>
```

Reply:
That  works too - it would be good to get this PR across the finish line.
We can apply what you did here to make the Chip component even better as Alex suggested --  thanks for the head start.

Reply:
Yes that’s fine, @AnnMarieW let’s just do it before we make a release and it becomes a breaking change.